src/ir_Corona.cpp

```diff
@@ -0,0 +1,569 @@
+/// @param[in] state An array of bytes containing the section
+/// @param[in] pos where to start in the state array
+/// @param[in] section which section to work with
+///            Used to get the section byte, and is validated against pos
+/// @return true if section is valid, otherwise false
+bool IRCoronaAc::validSection(const uint8_t state[], const uint16_t pos,
+                              const uint8_t section) {
+  // sanity check, pos must match section, section 4 is at pos 0
+  if ((section % kCoronaAcSections) * kCoronaAcSectionBytes != pos)
+    return false;
+  // all individual sections has the same prefix
+  if (state[pos + kCoronaAcSectionB0Pos] != kCoronaAcSectionB0) {
+    DPRINT("State ");
+    DPRINT(pos);
+    DPRINT(" expected 0x28 was ");
+    DPRINTLN(uint64ToString(state[pos + kCoronaAcSectionB0Pos], 16));
+    return false;
+  }
+  if (state[pos + kCoronaAcSectionB1Pos] != kCoronaAcSectionB1) {
+    DPRINT("State ");
+    DPRINT(pos + kCoronaAcSectionB1Pos);
+    DPRINT(" expected 0x61 was ");
+    DPRINTLN(uint64ToString(state[pos + kCoronaAcSectionB1Pos], 16));
+    return false;
+  }
+
+  // checking section byte
+  if (state[pos + kCoronaAcSectionLabelPos] != getSectionByte(section)) {
+    DPRINT("check 2 not matching, got ");
+    DPRINT(uint64ToString(state[pos + kCoronaAcSectionLabelPos], 16));
+    DPRINT(" expected ");
+    DPRINTLN(uint64ToString(getSectionByte(section), 16));
+    return false;
+  }
+
+  // checking inverts
+  uint8_t d0invinv = ~state[pos + kCoronaAcSectionD0InvPos];
+  if (state[pos + kCoronaAcSectionD0Pos] != d0invinv) {
+    DPRINT("inverted 3 - 4 not matching, got ");
+    DPRINT(uint64ToString(state[pos + kCoronaAcSectionD0Pos], 16));
+    DPRINT(" vs ");
+    DPRINT(uint64ToString(state[pos + kCoronaAcSectionD0InvPos], 16));
+    DPRINT(" expected ");
+    DPRINTLN(uint64ToString(~state[pos + kCoronaAcSectionD0Pos] & 0xff, 16));
+    return false;
+  }
+  uint8_t d1invinv = ~state[pos + kCoronaAcSectionD1InvPos];
+  if (state[pos + kCoronaAcSectionD1Pos] != d1invinv) {
+    DPRINT("inverted 5 - 6 not matching, got ");
+    DPRINT(uint64ToString(state[pos + kCoronaAcSectionD1Pos], 16));
+    DPRINT(" vs ");
+    DPRINT(uint64ToString(state[pos + kCoronaAcSectionD1InvPos], 16));
+    DPRINT(" expected ");
+    DPRINTLN(uint64ToString(~state[pos + kCoronaAcSectionD1Pos] & 0xff, 16));
+    return false;
+  }
+  return true;
+}
+
+/// Calculate and set the check values for the internal state.
+/// @param[in,out] data the array to be modified
+void IRCoronaAc::checksum(uint8_t* data) {
+  uint8_t pos;
+  for (uint8_t section = 0; section < kCoronaAcSections; section++) {
+    pos = section * kCoronaAcSectionBytes;
+    data[pos + kCoronaAcSectionB0Pos] = kCoronaAcSectionB0;
+    data[pos + kCoronaAcSectionB1Pos] = kCoronaAcSectionB1;
+    data[pos + kCoronaAcSectionLabelPos] = getSectionByte(section);
+    data[pos + kCoronaAcSectionD0InvPos] =
+      ~data[pos + kCoronaAcSectionD0Pos];
```

Comment:
Fold back into a single line, or indent by 4 spaces. (i.e. two more) Ditto for others.